radix-engine-common/src/math/decimal.rs

```diff
@@ -740,36 +740,51 @@ impl FromStr for Decimal {
     type Err = ParseDecimalError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let tens = I192::TEN;
         let v: Vec<&str> = s.split('.').collect();
 
-        let mut int = match I192::from_str(v[0]) {
+        let integer_part = match I192::from_str(v[0]) {
             Ok(val) => val,
-            Err(_) => return Err(ParseDecimalError::InvalidDigit),
+            Err(err) => match err {
+                ParseI192Error::NegativeToUnsigned => unreachable!("Not possible to be thrown by from_str"),
```

Comment:
Minor: `unreachable!("NegativeToUnsigned is only for parsing unsigned type, not I192"]`